### PR TITLE
Fix#788 ios show int fc

### PR DESCRIPTION
**Pull request by diepes**
##### ISSUE TYPE

 - Bugfix Pull Request

##### SUMMARY

Fix's #788 
templates/cisco_ios_show_interfaces_status.textfsm

be able to match fc interfaces, and add new {MODE} for fc interfaces.

```

```

### CODE REVIEW

File: templates/cisco_ios_show_interfaces_status.textfsm [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
 Value TYPE (.*)
+Value MODE (\S+)
 
 Start
-  ^Port -> Begin
-  # Capture time-stamp if vty line has command time-stamping turned on
   ^Load\s+for\s+
+  # Capture time-stamp if vty line has command time-stamping turned on
   ^Time\s+source\s+is
+  ^-{50,99}\s*$$
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_interfaces_status.textfsm:14]:
```suggestion
  ^-+ -> Error
```

File: templates/cisco_ios_show_interfaces_status.textfsm [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
+  ^Port\s+Name\s+Status\s+Vlan\s+Duplex\s+Speed\s+Type -> Interfaces
+  ^\s*$$
+  ^.* -> Error
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_interfaces_status.textfsm:17]:
```suggestion
  ^. -> Error
```

File: templates/cisco_ios_show_interfaces_status.textfsm [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
 
-Begin
+Interfaces
+  #Match fc...
+  ^\s*${PORT}\s+is\s+${STATUS}\s+Port\s+mode\s+is\s+${MODE}\s*$$ -> Record
   ^\s*${PORT}\s+${STATUS}\s+${VLAN}\s+${DUPLEX}\s+${SPEED}\s*${TYPE}$$ -> Record
   ^\s*${PORT}\s+${NAME}\s+${STATUS}\s+${VLAN}\s+${DUPLEX}\s+${SPEED}\s*${TYPE}$$ -> Record
+  ^-{50,99}\s*$$
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_interfaces_status.textfsm:24]:
```suggestion
  ^-+
```

File: templates/cisco_ios_show_interfaces_status.textfsm [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+  ^\s*$$
+  ^.* -> Error
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_interfaces_status.textfsm:26]:
```suggestion
  ^. -> Error
```

File: templates/cisco_ios_show_interfaces_status.textfsm [hunk 5]
```diff
@@ -1,17 +1,26 @@
 Value PORT (\S+)
 Value NAME (.+?)
-Value STATUS (err-disabled|disabled|connected|notconnect|inactive)
+Value STATUS (err-disabled|disabled|connected|notconnect|inactive|up|down)
 Value VLAN (\S+)
 Value DUPLEX (\S+)
 Value SPEED (\S+)
 Value TYPE (.*)
+Value MODE (\S+)
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_interfaces_status.textfsm:8]: Why was `MODE` added? I don't see it in the header of the output.

File: tests/cisco_ios/show_interfaces_status/cisco_ios_show_interfaces_status2.raw [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
+--------------------------------------------------------------------------------
+Port          Name               Status    Vlan      Duplex  Speed   Type
+--------------------------------------------------------------------------------
+fc2/1 is up    Port mode is NP
+fc2/2 is up    Port mode is NP
+fc2/3 is up    Port mode is NP
+fc2/4 is up    Port mode is NP
+fc2/5 is down (SFP not present)
```
Comment (FragmentedPacket) [inline tests/cisco_ios/show_interfaces_status/cisco_ios_show_interfaces_status2.raw:9]: We should make a new capture group for L9.

```
  ^\s*${PORT}\s+is\s+${STATUS}\s+\(${TYPE}\)$$
```

**Review by FragmentedPacket — CHANGES_REQUESTED** [review]

File: templates/cisco_ios_show_interfaces_status.textfsm (hunk 5 above)
Comment (diepes) [reply templates/cisco_ios_show_interfaces_status.textfsm]: Renamed to FC_MODE  it indicates the MODE of the Fiber channel e.g. "FP" , not the same as type, as TYPE refers to the physical port type.

File: templates/cisco_ios_show_interfaces_status.textfsm (hunk 4 above)
Comment (diepes) [reply templates/cisco_ios_show_interfaces_status.textfsm]: Done.

File: tests/cisco_ios/show_interfaces_status/cisco_ios_show_interfaces_status2.raw (hunk 6 above)
Comment (diepes) [reply tests/cisco_ios/show_interfaces_status/cisco_ios_show_interfaces_status2.raw:9]: Good catch, added a match for the FC DOWN.  Did not use TYPE as TYPE is the L1(Physical) interface/sfp type

File: templates/cisco_ios_show_interfaces_status.textfsm (hunk 5 above)
Comment (diepes) [reply templates/cisco_ios_show_interfaces_status.textfsm]: To show the FC port mode config,  renamed to FC_MODE.

**Comment by diepes** [comment]
Changes implemented, new Value  FC_MODE, captures the FC mode "NP" in example, not the same as TYPE the L1 - Interface type.

**Review by FragmentedPacket — APPROVED** [review]
Thanks for the great work!

**Pull request merged**